### PR TITLE
fix: refuse schedule compaction tasks if there is no slot

### PR DESCRIPTION
**Pull request by tedxu**
See #37621

### CODE REVIEW

File: internal/datacoord/compaction.go [hunk 1]
```diff
@@ -592,49 +622,51 @@ func (c *compactionPlanHandler) createCompactTask(t *datapb.CompactionTask) (Com
...
-		}
+	nodeID, useSlot := pickAnyNode(slots, t)
+	if nodeID == NullNodeID {
+		log.Info("compactionHandler cannot find datanode for compaction task",
+			zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("type", t.GetTaskProto().GetType().String()), zap.String("vchannel", t.GetTaskProto().GetChannel()))
+		return NullNodeID
 	}
+	err := t.SetNodeID(nodeID)
+	if err != nil {
+		log.Info("compactionHandler assignNodeID failed",
```
Comment (XuanYang-cn) [inline internal/datacoord/compaction.go:638]: Please change log level to Warning

**Review by XuanYang-cn — COMMENTED** [review]
/lgtm

File: internal/datacoord/compaction.go (hunk 1 above)
Comment (tedxu) [reply internal/datacoord/compaction.go:638]: This is copied from previous impl, will change it later.

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**